models/ed/R/met2model.ED2.R

```diff
@@ -176,16 +179,32 @@ met2model.ED2 <- function(in.path, in.prefix, outfolder, start_date, end_date, l
     useCO2 <- is.numeric(CO2)
 
     ## convert time to seconds
-    sec <- udunits2::ud.convert(sec, unlist(strsplit(nc$dim$time$units, " "))[1], "seconds")
+    sec <- udunits2::ud.convert(tdays, unlist(strsplit(nc$dim$time$units, " "))[1], "seconds")
 
     ncdf4::nc_close(nc)
 
-    dt <- PEcAn.utils::seconds_in_year(year, leap_year) / length(sec)
+    # `dt` is the met product time step in seconds. We calculate it here as
+    # `sec[i] - sec[i-1]` for all `[i]`. For a properly formatted product, the
+    # timesteps should be regular, so there is a single, constant difference. If
+    # that's not the case, we throw an informative error.
+    #
+    # `drop` here simplifies length-1 arrays to vectors. Without it, R will
+    # later throw an error about "non-conformable arrays" when trying to add a
+    # length-1 array to a vector.
+    dt <- drop(unique(diff(sec)))
```

Comment:
for reasons I don't 100% understand (roundoff?) dt doesn't always have a single unique value in all met products. You might be better off with a round(mean(diff or round(median(diff instead

Reply:
Good to know, thanks! See https://github.com/PecanProject/pecan/pull/2734/commits/ef5244adbc3fdd54eca691bfd77264d2c2b06866. I changed it so we try to grab the unique value first, and if that fails, we throw a warning and use the rounded mean instead.